### PR TITLE
Create cloud_grains_present state

### PR DESCRIPTION
**Pull request by arbulu89**
New salt state for https://github.com/SUSE/habootstrap-formula/pull/58

It basically assigns the required cloud information to grains to be used later. This is the unique way to use runtime variables in salt in the same execution.

Fix for: https://github.com/SUSE/habootstrap-formula/issues/57

### CODE REVIEW

**Review by MalloZup — COMMENTED** [review]
to me it looks more complicating the code instead of solving issue.

I'm referring on the comment I left on https://github.com/SUSE/habootstrap-formula/issues/57

**Comment by arbulu89** [comment]
@MalloZup It's not more complicated. We had the same `http` calls before, exactly the same. The unique difference is that the code execution has been moved to salt states instead of `jinja` code, what is not really recommended.
Salt code in `sls` formats has some limitations, and due them, we are forced to create code in `python`. I would like to have this implementation in the formula too, but it's not possible as far I have researched.

**Comment by MalloZup** [comment]
@arbulu89  yes I checked  the http request is moved. make sense WIll check during the day

File: salt/states/crmshmod.py [hunk 1]
```diff
@@ -548,3 +548,50 @@ def cluster_op_defaults_present(
     ret['comment'] = 'Cluster op_defaults configured'
     ret['result'] = True
     return ret
+
+
+def cloud_grains_present(
+        name):
+    """
+    Set the required cloud providers data in the grains
```
Comment (MalloZup) [inline salt/states/crmshmod.py:556]: can we add why this is needed on highlevel manner?
why do we need to set them?

**Review by MalloZup — APPROVED** [review]
ok only minor comment on UX

File: salt/states/crmshmod.py [hunk 2]
```diff
@@ -548,3 +548,50 @@ def cluster_op_defaults_present(
     ret['comment'] = 'Cluster op_defaults configured'
     ret['result'] = True
     return ret
+
+
+def cloud_grains_present(
```
Comment (Simranpal) [inline salt/states/crmshmod.py:553]: should this not be named like `set_cloud_grains`

**Review by Simranpal — APPROVED** [review]
Thanks @arbulu89 just a comment about naming. This approach is aligns better than jinja salt calls

File: salt/states/crmshmod.py (hunk 2 above)
Comment (arbulu89) [reply salt/states/crmshmod.py:553]: Why do you prefer `set_cloud_grains` over `cloud_grains_present`?
This is a salt state, and as convention, the states are named with this syntax, saying which will be the state of the provisioning after the execution.
If you check the rest of the file, the rest of the methods are following the same syntax rules.

Comment (Simranpal) [reply salt/states/crmshmod.py:553]: yes, you are right, I forgot this was salt state name

**Pull request merged**